templates/default/bump_environments.rb.erb

```diff
@@ -0,0 +1,101 @@
+#!/opt/chef/embedded/bin/ruby
+
+# Given a chef repo name, a cookbook name, a version to bump environments to,
+# and a comma-separated list of environments passed as environment variables,
+# bumps the cookbook to the given version in each of the environments (or all
+# of them if no environments are specified) and creates a PR for it on Github.
+# Also optionally takes a link to the PR that triggered the version bump.
+
+require 'git'
+require 'json'
+require 'octokit'
+require 'set'
+
+# TODO: Validate input
+
+DEFAULT_CHEF_ENVS = <%= @default_environments %>.freeze
+DEFAULT_CHEF_ENVS_WORD = '<%= @default_environments_word %>'.freeze
+ALL_CHEF_ENVS_WORD = '<%= @all_environments_word %>'.freeze
+CHEF_REPO = '<%= @chef_repo %>'.freeze
+
+cookbook = ENV['cookbook']
+version = ENV['version']
+pr_link = ENV['pr_link']
+
+# Only bump in the chef environments specified
+chef_envs = ENV['envs'].split(',').to_set
+
+# If 'default environments' word is found, replace it with the default envs.
+if chef_envs.include?(DEFAULT_CHEF_ENVS_WORD)
+  chef_envs.delete(DEFAULT_CHEF_ENVS_WORD)
+  chef_envs += DEFAULT_CHEF_ENVS
+end
+
+# Only note that we're using the defaults if we're using nothing else.
+is_default_envs = chef_envs == DEFAULT_CHEF_ENVS.to_set
+
+# If 'all environments' word is found, bump all environments.
+if chef_envs.include?(ALL_CHEF_ENVS_WORD)
+  is_all_envs = true
+  chef_env_files = Dir.glob('environments/*.json')
+  chef_envs = chef_env_files.map do |f|
+    f.slice(%r{environments/(.*)\.json}, 1)
+  end.to_set
+else
+  chef_env_files = chef_envs.map { |e| "environments/#{e}.json" }
+end
+
+# Create a new branch
+Git.configure do |config|
+  config.binary_path = '<%= @git_path %>'
+end
+git = Git.open('.')
+git.branch('master').checkout
+branch_hash = if is_all_envs
+                'in-all-envs'
+              elsif is_default_envs
+                'in-default-envs'
+              else
+                ''
+              end
+# Add a 10-digit hash of the version and envs so the branch names are unique.
+branch_hash += "-#{(chef_envs.to_a << version).join(',').hash % 10_000_000_000}"
+git_branch = "jenkins/bump-#{cookbook}-to-#{version}-#{branch_hash}"
```

Comment:
I think this should be shortened to `jenkins/#{cookbook}-#{version}-#{branch_hash}`